java/feathub-connectors/flink-connectors/flink-connector-redis/src/main/java/com/alibaba/feathub/flink/connectors/redis/sink/RedisSinkFunction.java

```diff
@@ -0,0 +1,234 @@
+ */
+public class RedisSinkFunction extends RichSinkFunction<RowData> {
+
+    private final String host;
+    private final int port;
+    private final String username;
+    private final String password;
+    private final int dbNum;
+
+    private final byte[] keyPrefix;
+    private final int keyFieldIndex;
+    private final int timestampFieldIndex;
+
+    private final byte[] evalScript;
+    private byte[] evalScriptSHA;
+
+    private transient ByteBuffer indexBuffer;
+    private transient ByteBuffer timestampBuffer;
+
+    private transient Jedis jedis;
+
+    public RedisSinkFunction(ReadableConfig config, ResolvedSchema schema) {
+        this.host = config.get(HOST);
+        this.port = config.get(PORT);
+        this.username = config.get(USERNAME);
+        this.password = config.get(PASSWORD);
+        this.dbNum = config.get(DB_NUM);
+
+        this.keyPrefix = getKeyPrefix(config.get(NAMESPACE));
+        this.keyFieldIndex = getKeyFieldIndex(config, schema);
+        this.timestampFieldIndex = getTimestampFieldIndex(config, schema);
+
+        this.evalScript = getEvalScript(timestampFieldIndex);
+
+        validateConfigAndSchema(config, schema, keyFieldIndex, timestampFieldIndex);
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        super.open(parameters);
+
+        this.indexBuffer = ByteBuffer.allocate(4).order(ByteOrder.BIG_ENDIAN);
+        this.timestampBuffer = ByteBuffer.allocate(8).order(ByteOrder.BIG_ENDIAN);
+
+        jedis = new Jedis(host, port);
+        if (!StringUtils.isNullOrWhitespaceOnly(username)) {
+            jedis.auth(username, password);
+        } else if (!StringUtils.isNullOrWhitespaceOnly(password)) {
+            jedis.auth(password);
+        }
+        jedis.select(dbNum);
+        if (timestampFieldIndex >= 0) {
+            evalScriptSHA = jedis.scriptLoad(this.evalScript);
+        }
+    }
+
+    @Override
+    public void invoke(RowData data, Context context) {
+        byte[] originalKey = data.getBinary(keyFieldIndex);
+        byte[] key = new byte[keyPrefix.length + originalKey.length];
+        System.arraycopy(keyPrefix, 0, key, 0, keyPrefix.length);
+        System.arraycopy(originalKey, 0, key, keyPrefix.length, originalKey.length);
+
+        Map<byte[], byte[]> value = new HashMap<>();
+        int arity = data.getArity();
+        for (int i = 0; i < arity; i++) {
+            if (timestampFieldIndex == i) {
+                // timestamp values have specific serialization strategy
+                continue;
+            }
+
+            if (keyFieldIndex == i) {
+                // don't store key values
+                continue;
+            }
+
+            indexBuffer.putInt(0, i);
+            value.put(indexBuffer.array(), data.getBinary(i));
+        }
+
+        if (timestampFieldIndex < 0) {
+            if (value.isEmpty()) {
```

Comment:
Since `validateConfigAndSchema()` should guarantee there is at least one extra column in addition to timestamp and key columns, can value still be empty here?